classes/core.php

```diff
@@ -1447,9 +1470,40 @@ static public function run_calculation($value, $field, $form)
 			return new WP_Error($field['ID'] . '-calculation',
 				__('Calculation is invalid (division by zero)', 'caldera-forms'));
 		}
+	
+		//If PHP version is less than 7.2.0, continue using old function
+		if(phpversion() < '7.2.0'){
+
+			$total_function = create_function(null, 'return ' . $formula . ';');
+			$total = $total_function();
+
+		} else { //avoid using create_function() when using PHP version >= 7.2.0
+
+			//Check if the formula use a symbol or function not supported by the parser but declared as supported at https://calderaforms.com/doc/calculation-fields/
+			$deprecated = self::check_deprecated_math_functions( $formula );
 
-		$total_function = create_function(null, 'return ' . $formula . ';');
-		$total = $total_function();
+			//Use parser if no deprecated caught
+			if( $deprecated === false ){ 
+				//Initiate parser
+				$parser = new FormulaParser($formula);
+				$result = $parser->getResult();
+				//Get result if parser returns a correct status
+				if($result[0] === "done"){ 
+					$total = $result['1'];
+				} else if($result[0] === "error") {  //Return an error if parser returned an error status
+					return new WP_Error($field['ID'] . '-calculation', __('Deprecated operator or symbol in calculation for PHP version >= 7.2 requirements', 'caldera-forms'));
+				}
+				
+			} else { //Return an error with the deprecated symbol or function caught
```

Comment:
@New0 I don't agree with this decision. I think we should still run the submission. If they are an admin, we should attach this warning to the form notices. If not, report nothing.

Check out here where I did the notices differnetly based on if the current user can fix the issue or not:

https://github.com/CalderaWP/Caldera-Forms/pull/3202/files#diff-cda146b98bb4831185232d8634a1f85eR109

See here for example of changing admin notices. I suspect this will require its own class
https://github.com/CalderaWP/Caldera-Forms/blob/master/includes/cf-pro-client/classes/hooks.php#L37-L38